### PR TITLE
feat: Implement PATCH device command

### LINKED ISSUE #650: Implement testcase for PATCH device command
# 🚀 Feature Request

### Relevant Package [**REQUIRED**]
This feature request is for metadata

### Description [**REQUIRED**]
According to https://docs.edgexfoundry.org/2.2/design/adr/core/0021-Device-Profile-Changes/ & https://github.com/edgexfoundry/edgex-go/issues/3925
The PATCH device command implement tests would be added to metadata.

### Describe the solution you'd like
 If you have a solution in mind, please describe it.

### Describe alternatives you've considered
 Have you considered any alternative solutions or workarounds?

### PR DESCRIPTION
**Pull request by jumpingliu**
fix #650

Signed-off-by: melody <melody@iotechsys.com>

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [ ] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PATCH-Negative-DeviceCommand.robot [hunk 1]
```diff
@@ -13,30 +13,34 @@ ${LOG_FILE_PATH}  ${WORK_DIR}/TAF/testArtifacts/logs/core-metadata-deviceprofile
 *** Test Cases ***
 ErrProfileCommandPATCH001 - Update deviceCommand with device profile name validation error
     # non-existent device profile name
-    When Update deviceCommand with Non-existent device profile name
+    Given Generate a profile and command sample for updating
```
Comment (cherrycl) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PATCH-Negative-DeviceCommand.robot:16]: Create profile is unnecessary for non-existent profile name.

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PATCH-Positive-DeviceCommand.robot [hunk 2]
```diff
@@ -4,41 +4,52 @@ Resource     TAF/testCaseModules/keywords/core-metadata/coreMetadataAPI.robot
...
+    Set To Dictionary  ${commandUpdate}[0]  profileName=Test-Profile-1
+    Set To Dictionary  ${commandUpdate}[1]  profileName=Test-Profile-2
+    Set To Dictionary  ${commandUpdate}[2]  profileName=Test-Profile-3
 
+Profile ${profile_name} command isHidden Should Be Updated
+    Query device profile by name  ${profile_name}
+    Run Keyword IF  "${profile_name}" == "Test-Profile-1"
+    ...             Should Be Equal  ${content}[profile][deviceCommands][0][isHidden]  ${false}
+    ...    ELSE IF  "${profile_name}" == "Test-Profile-2"  Should Be Equal  ${content}[profile][deviceCommands][0][isHidden]  ${true}
+    ...    ELSE IF  "${profile_name}" == "Test-Profile-3"  Should Be Equal  ${content}[profile][deviceCommands][0][isHidden]  ${true}
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PATCH-Positive-DeviceCommand.robot:55]: `if else` is unnecessary in this case if `true/false` is a variable in the keyword name
```suggestion
Profile ${profile_name} command isHidden Should Be ${bool}
    Query device profile by name  ${profile_name}
    Should Be Equal  ${content}[profile][deviceCommands][0][isHidden]  ${bool}
```

**Review by jinlinGuan — CHANGES_REQUESTED** [review]

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PATCH-Negative-DeviceCommand.robot (hunk 1 above)
Comment (jumpingliu) [reply TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PATCH-Negative-DeviceCommand.robot]: updated

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PATCH-Positive-DeviceCommand.robot (hunk 2 above)
Comment (jumpingliu) [reply TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PATCH-Positive-DeviceCommand.robot]: updated

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PATCH-Positive-DeviceCommand.robot [hunk 3]
```diff
@@ -4,41 +4,49 @@ Resource     TAF/testCaseModules/keywords/core-metadata/coreMetadataAPI.robot
...
+    Given Generate a profile and command sample for updating
+    When Update command ${commandUpdate}
     Then Should Return Status Code "207"
     And Should Return Content-Type "application/json"
     And Item Index All Should Contain Status Code "200"
     And Response Time Should Be Less Than "${default_response_time_threshold}"ms
-    And deviceCommand Should Be Updated
-    [Teardown]  Delete Device Profile By Name
+    And Profile Test-Profile-1 command isHidden Should Be ${false}
+    [Teardown]  Delete Device Profile By Name  Test-Profile-1
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PATCH-Positive-DeviceCommand.robot:23]:
```suggestion
    [Teardown]  Delete Device Profile By Name  ${test_profile}
```

**Review by jinlinGuan — CHANGES_REQUESTED** [review]

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PATCH-Positive-DeviceCommand.robot (hunk 3 above)
Comment (jumpingliu) [reply TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PATCH-Positive-DeviceCommand.robot]: updated

**Review by jinlinGuan — APPROVED** [review]
LGTM

**Pull request merged**